addons/event_crm/models/event_event.py

```diff
@@ -23,3 +25,38 @@ def _compute_lead_count(self):
         mapped_data = {event.id: count for event, count in lead_data}
         for event in self:
             event.lead_count = mapped_data.get(event.id, 0)
+
+    def action_generate_leads(self):
+        """ Re-generate leads based on event.lead.rules.
+        The method is ran synchronously if there is a low amount of registrations, otherwise it
+        goes through a CRON job that runs in batches. """
+
+        if not self.env.user.has_group('event.group_event_manager'):
+            raise UserError(_("Only Event Managers are allowed to re-generate all leads."))
+
+        self.ensure_one()
+        registrations_count = self.env['event.registration'].search_count(
```

Comment:
Shouldn't we exclude draft and canceled registrations ? Not sure for draft, but as now draft is used for unconfirmed registrations linked to SO, would makes sense to exclude them.

Reply:
They are not created in cancel mode ... now it is only in confirm. Triggers are: new attendee (create), attendee confirmed (state to open, now real default as draft is just a temporary one for sales), attendee done. So I think we should probably at least filter out canceled attendees, and probably draft attendees also.